integration/e2e/e2e_test.go

```diff
@@ -914,3 +925,98 @@ func hashFile(file string) string {
 func chaincodeContainerNameFilter(n *nwo.Network, chaincode nwo.Chaincode) string {
 	return fmt.Sprintf("^/%s-.*-%s-%s$", n.NetworkID, chaincode.Label, hashFile(chaincode.PackageFile))
 }
+
+type channelList struct {
+	SystemChannel *channelInfoShort  `json:"systemChannel"`
+	Channels      []channelInfoShort `json:"channels"`
+}
+
+type channelInfoShort struct {
+	Name string `json:"name"`
+	URL  string `json:"url"`
+}
+
+func channelParticipationList(n *nwo.Network, o *nwo.Orderer, expectedChannels []string, systemChannel ...string) {
+	authClient, unauthClient := OrdererOperationalClients(n, o)
+	listChannelsURL := fmt.Sprintf("https://127.0.0.1:%d/participation/v1/channels", n.OrdererPort(o, nwo.OperationsPort))
+
+	By("using the channel participation API to list the channels")
+	body := getBody(authClient, listChannelsURL)()
+	list := &channelList{}
+	err := json.Unmarshal([]byte(body), list)
+	Expect(err).NotTo(HaveOccurred())
+
+	Expect(*list).To(MatchFields(IgnoreExtras, Fields{
+		"Channels":      channelsMatcher(expectedChannels),
+		"SystemChannel": systemChannelMatcher(systemChannel...),
+	}))
+
+	By("listing the channels without a client cert")
+	resp, err := unauthClient.Get(listChannelsURL)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(resp.StatusCode).To(Equal(http.StatusUnauthorized))
+
+	for _, channel := range list.Channels {
+		By("listing the details for channel " + channel.Name)
+		channelParticipationListOne(n, o, channel)
+	}
+}
+
+func channelsMatcher(channels []string) types.GomegaMatcher {
+	if len(channels) == 0 {
+		return BeEmpty()
+	}
+	matchers := make([]types.GomegaMatcher, len(channels))
+	for i, channel := range channels {
+		matchers[i] = channelInfoShortMatcher(channel)
+	}
+	return ConsistOf(matchers)
+}
+
+func systemChannelMatcher(systemChannel ...string) types.GomegaMatcher {
+	if len(systemChannel) == 0 {
+		return BeNil()
+	}
+	return PointTo(channelInfoShortMatcher(systemChannel[0]))
+}
+
+func channelInfoShortMatcher(channel string) types.GomegaMatcher {
+	return MatchFields(IgnoreExtras, Fields{
+		"Name": Equal(channel),
+		"URL":  Equal(fmt.Sprintf("/participation/v1/channels/%s", channel)),
+	})
+}
+
+type channelInfo struct {
+	Name            string `json:"name"`
+	URL             string `json:"url"`
+	Status          string `json:"status"`
+	ClusterRelation string `json:"clusterRelation"`
+}
```

Comment:
Where is the Height? 
And why repeat the definition of orderer/types/ChannelInfo?

Reply:
Mirroring the types here provides us an automated way to notice any breaking changes that will affect consumers. For this, I only included the fields I could test the exact values. We could include Height but it would require some additional logic to determine the expected value (or we could just check that it's set to a non-empty value).